### PR TITLE
EE-8277: Introduce a POST endpoint.

### PR DESCRIPTION
**Pull request by Tim-Hammonds**

### CODE REVIEW

File: src/main/resources/logback-spring.xml [hunk 1]
```diff
@@ -39,7 +39,7 @@
                 <provider class="net.logstash.logback.composite.loggingevent.LoggingEventPatternJsonProvider">
                     <pattern>
                         {
-                        "message": "%replace(%msg){'([A-Z]{2}[0-9]{3})[0-9]{3}[A-D]','$1****'}"
+                        "message": "%replace(%msg){'([A-Za-z]\\s*[A-Za-z]\\s*[0-9]\\s*[0-9]\\s*[0-9]\\s*)[0-9]\\s*[0-9]\\s*[0-9]\\s*[A-Da-d]','$1****'}"
```
Comment (zuzak) [inline src/main/resources/logback-spring.xml]: `/(\w{2}\d)\d{5}[A-Da-d]/`?
Or if whitespace is important: `((\w\s*){2}\d\s*)(\d\s*){5}[A-Da-d]`

Comment (mikehalmaho) [reply src/main/resources/logback-spring.xml]: Does \w include 0-9?  They are omitted on purpose (this is a nino).  However I think the whitespace change is better.

File: src/main/java/uk/gov/digital/ho/pttg/api/IncomeDataRequest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+
+import java.time.LocalDate;
+
+@AllArgsConstructor
+@Getter
+@Accessors(fluent = true)
+class IncomeDataRequest {
```
Comment (mikehalmaho) [inline src/main/java/uk/gov/digital/ho/pttg/api/IncomeDataRequest.java]: Is "Data" redundant?

File: src/test/java/uk/gov/digital/ho/pttg/api/HmrcResourceTest.java [hunk 3]
```diff
@@ -39,12 +42,15 @@
     @Mock private IncomeSummary mockIncomeSummary;
     @Mock private Appender<ILoggingEvent> mockAppender;
 
+    @Captor private ArgumentCaptor<Individual> captorIndividual;
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/pttg/api/HmrcResourceTest.java]: Is individualCaptor better?  Yeah I'm clutching at straws here :)

**Pull request merged**